### DISCUSSION TITLE
Sync from Mac to iPhone

### DISCUSSION THREAD

**Discussion by odedia** [RhetTbull/osxphotos#1502] [Ideas]
Apple's "manual" sync feature from Apple Photos to iOS is reaching a point where it's so buggy, it's just not usable at all. 
I have a 1TB iPhone that can easily hold iPhone-optimized JPEGs and all my videos (some may need conversion to play on iPhone).

Plus - Apple are so nice that they don't let you sync your live photos back to the iPhone:

![5E9091AC-BB79-4058-8A98-E79F33F4BB1D_1_105_c](https://github.com/RhetTbull/osxphotos/assets/2992180/cdd34a89-a70c-43c7-8a76-e550c592183a)

I don't even know if it's feasible, but given that options such as iMazing exist, I wonder if it would be possible to have osxphotos "sync" converted photos and videos (or originals if that's easier) directly to an iPhone, with the option to `--cleanup`, thus making it some sort of a sync tool back to the iPhone?

Thanks again for this great tool.

**Comment by RhetTbull** [comment]
This *might* be possible using something like [iFuse](https://github.com/libimobiledevice/ifuse) to mount the phone's filesystem to the Mac and let osxphotos run on it. Fixing things like the Live Photos would require directly editing the Photos database on the phone which is tricky. iFuse was *very* slow last I tried it. This isn't something I have time to work on (probably not this year to be honest) but it's an interesting idea. Feel free to open an issue with this as a feature request for tracking. Someone else with a similar use case might come along and work on it. It's a non-trivial problem given how opaque Apple is and Apple really wants to drive you to use iCloud. I'm an iCloud user so this isn't something I need (and hence lower on my priority list).

**Reply by RhetTbull** [reply]
Also, it might make sense to use some sort of third party library app on the phone and then write an osxphotos script that syncs to that. Probably easier than messing with the black box that is iPhone.
